File: Medium/2026-03-20-3567-Minimum-Absolute-Difference-in-Sliding-Submatrix/solution.js
var minAbsDiffInSubmatrix = function(grid, k) {
    const m = grid.length;
    const n = grid[0].length;
    const ans = Array.from({ length: m - k + 1 }, () => Array(n - k + 1).fill(0));

    for (let i = 0; i <= m - k; i++) {
        for (let j = 0; j <= n - k; j++) {
            const values = new Set();
            for (let x = i; x < i + k; x++) {
                for (let y = j; y < j + k; y++) {
                    values.add(grid[x][y]);
                }
            }
            const sortedValues = Array.from(values).sort((a, b) => a - b);
            let minDiff = Infinity;

            for (let l = 1; l < sortedValues.length; l++) {
                minDiff = Math.min(minDiff, sortedValues[l] - sortedValues[l - 1]);
            }

            ans[i][j] = minDiff === Infinity ? 0 : minDiff;
        }
    }

    return ans;
};

File: Medium/2026-03-20-3567-Minimum-Absolute-Difference-in-Sliding-Submatrix/solution.java
import java.util.*;

public class Solution {
    public int[][] minimumAbsDifference(int[][] grid, int k) {
        int m = grid.length, n = grid[0].length;
        int[][] ans = new int[m - k + 1][n - k + 1];

        for (int i = 0; i <= m - k; i++) {
            for (int j = 0; j <= n - k; j++) {
                Set<Integer> distinctValues = new HashSet<>();
                for (int x = i; x < i + k; x++) {
                    for (int y = j; y < j + k; y++) {
                        distinctValues.add(grid[x][y]);
                    }
                }
                List<Integer> valuesList = new ArrayList<>(distinctValues);
                Collections.sort(valuesList);
                ans[i][j] = findMinAbsDifference(valuesList);
            }
        }
        return ans;
    }

    private int findMinAbsDifference(List<Integer> values) {
        if (values.size() < 2) return 0;
        int minDiff = Integer.MAX_VALUE;
        for (int i = 1; i < values.size(); i++) {
            minDiff = Math.min(minDiff, values.get(i) - values.get(i - 1));
        }
        return minDiff;
    }
}